### ISSUE TITLE
Prototype ML Pipeline

### ISSUE THREAD

**Issue by marcos-j-salazar** [marcos-j-salazar/triage-copilot#2]
**Goal:** Create a notebook to test our TF-IDF + LogisticRegression pipeline, catch any data leakage issues during the train/test split, and decide how we want to handle categories vs. subcategories.

**The notebook must:**

- Load the seed dataset (training_phrases.csv) containing 100 bilingual source phrases.
- Execute a strict train/test split on the source phrases prior to any text augmentation to prevent near-duplicate data from bleeding across sets.
- Train a unified TfidfVectorizer and LogisticRegression pipeline.
- Restrict the model predictions to the 6 primary categories.
- Conduct EDA on the source phrases to identify distribution gaps and produce a concrete, data-driven recommendation for scaling the dataset.

**Acceptance Criteria:**
The pipeline is fully implemented and mathematically sound. The notebook successfully prevents data leakage by augmenting post-split, demonstrating a realistic baseline accuracy gap. The EDA successfully isolates the main cause of the overfitting and provides a clear strategy for how to expand the training data in the next development phase.

**Comment by marcos-j-salazar** [comment]
Built the initial pipeline using 100 bilingual NSCC source phrases. Flagged a critical bug during evaluation: Train and Test accuracy both returned a perfect 1.000, which led me to suspect data leakage.

Attached below is a screenshot of the code which caused the leakge.

<img width="790" height="752" alt="Image" src="https://github.com/user-attachments/assets/2e4552ae-56a4-4506-ac61-7c3cde593a65" />

**Comment by marcos-j-salazar** [comment]
Confirmed data leakage by verifying that 42/46 test phrases had >0.9 similarity to training data. Fixed the architecture to split source phrases first, then augment. Leakage is now at 0/49. Metrics: Train 0.985, Test 0.449, Gap 0.536

<img width="817" height="611" alt="Image" src="https://github.com/user-attachments/assets/a9cc6d9d-4c2d-4d98-b4e2-27040d7c04df" />

**Comment by marcos-j-salazar** [comment]
The authentic 0.536 gap confirms what the EDA warned about: text augmentation isn't adding enough real phrasing diversity to our thin subcategories. Decided to restric the ML model to predict 6 primary categories only. Proposed a scoped LLM generation spike (GPT or Gemini) to target 15-20 distinct phrases per subcategory to fix the data gap.

**Comment by marcos-j-salazar** [comment]
https://github.com/marcos-j-salazar/triage-copilot/commit/efb982e873ed3cb8eb49a44ca366a8362c0d580a
